usaspending_api/download/filestreaming/download_generation.py

```diff
@@ -63,17 +67,55 @@ def generate_download(download_job: DownloadJob, origination: Optional[str] = No
     file_format = json_request.get("file_format")
     request_type = json_request.get("request_type")
 
-    span = tracer.current_span()
-    if span and request_type:
-        span.resource = request_type
-
     file_name = start_download(download_job)
+
+    with SubprocessTrace(
+        name=f"job.{JOB_TYPE}.generate_download_{request_type}",
+        kind=SpanKind.INTERNAL,
+        service="bulk-download",
+    ) as main_trace:
+        main_trace.set_attributes(
+            {
+                "service": "bulk-download",
+                "span_type": "Internal",
+                "job_type": str(JOB_TYPE),
+                "message": "Creating data archive files from the download job object",
+                # download job details
+                "download_job_id": str(download_job.download_job_id),
+                "download_job_status": str(download_job.job_status.name),
+                "download_file_name": str(download_job.file_name),
+                "download_file_size": download_job.file_size if download_job.file_size is not None else 0,
+                "number_of_rows": download_job.number_of_rows if download_job.number_of_rows is not None else 0,
+                "number_of_columns": (
+                    download_job.number_of_columns if download_job.number_of_columns is not None else 0
+                ),
+                "error_message": download_job.error_message if download_job.error_message else "",
+                "monthly_download": str(download_job.monthly_download),
+                "json_request": str(download_job.json_request) if download_job.json_request else "",
+                "file_name": str(file_name),
+            }
+        )
+
     working_dir = None
     try:
         if limit is not None and limit > MAX_DOWNLOAD_LIMIT:
+
+            with SubprocessTrace(
+                name=f"job.{JOB_TYPE}.generate_download_{request_type}",
+                kind=SpanKind.INTERNAL,
+                service="bulk-download",
+            ) as limit_exceeded:
+                limit_exceeded.set_attributes(
+                    {
+                        "message": "Unable to process this download because it includes more than the current limit of {MAX_DOWNLOAD_LIMIT} records",
```

Comment:
```suggestion
                        "message": f"Unable to process this download because it includes more than the current limit of {MAX_DOWNLOAD_LIMIT} records",
```